### PR TITLE
Limited allowed inputs for file_path parameter in upgrade_custom endpoint

### LINKED ISSUE #17603: Limit the allowed inputs in the `file_path` parameter of `upgrade_custom`
## Description

The [PUT /agents/upgrade_custom](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.agent_controller.put_upgrade_custom_agents) endpoint will undergo some breaking changes in this issue:
- https://github.com/wazuh/wazuh/issues/17450

The breaking changes will mainly be:
- `filepath` parameter renamed to `wpk_filename`.
- `wpk_filename` will expect a filename, not a full/relative path to the file.
- Only files under `/var/ossec/var/upgrade` can be used.

Until those changes are applied, a minor development will take place: the file specified under `file_path` must end with `.wpk`. Otherwise, it should be rejected by the API.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

### PR DESCRIPTION
**Pull request by EduLeon12**
|Related issue|
|---|
|Closes #17603 |

## Description

As explained in #17603, Changing the parameter to filename is a breaking change that will not be addressed at the moment.
To address the issue a minor development of validating the extension of the file has been developed and tested. 

## Tests

More Test Result and logs related to issue can be seen [here](https://github.com/wazuh/wazuh/issues/17603#issuecomment-1603249558).  

<details> <summary> Test validator  </summary>

```
(unit-test) eduardoleon@pop-os:~/git/wazuh$ pytest api/api/test/test_validator.py
==================================== test session starts ====================================
platform linux -- Python 3.9.9, pytest-7.0.1, pluggy-0.13.1
rootdir: /home/eduardoleon/git/wazuh/api
plugins: asyncio-0.18.1, aiohttp-1.0.4, trio-0.7.0, cov-3.0.0, anyio-3.6.2, metadata-2.0.2, html-3.0.0
asyncio: mode=auto
collected 176 items

api/api/test/test_validator.py ...................................................... [ 30%]
..................................................................................... [ 78%]
.....................................                                                 [100%]

============================== 176 passed, 3 warnings in 0.19s ==============================
```

</details>

<details> <summary> Api Test </summary>

```
(unit-test) eduardoleon@pop-os:~/git/wazuh$ pytest --disable-warnings  api/api/
==================================== test session starts ====================================
platform linux -- Python 3.9.9, pytest-7.0.1, pluggy-0.13.1
rootdir: /home/eduardoleon/git/wazuh/api
plugins: asyncio-0.18.1, aiohttp-1.0.4, trio-0.7.0, cov-3.0.0, anyio-3.6.2, metadata-2.0.2, html-3.0.0
asyncio: mode=auto
collected 540 items

api/api/controllers/test/test_active_response_controller.py .                         [  0%]
api/api/controllers/test/test_agent_controller.py ................................... [  6%]
.........                                                                             [  8%]
api/api/controllers/test/test_cdb_list_controller.py ......                           [  9%]
api/api/controllers/test/test_ciscat_controller.py .                                  [  9%]
api/api/controllers/test/test_cluster_controller.py ........................          [ 14%]
api/api/controllers/test/test_decoder_controller.py .......                           [ 15%]
api/api/controllers/test/test_default_controller.py .                                 [ 15%]
api/api/controllers/test/test_experimental_controller.py ...............              [ 18%]
api/api/controllers/test/test_manager_controller.py ..................                [ 21%]
api/api/controllers/test/test_mitre_controller.py .......                             [ 22%]
api/api/controllers/test/test_overview_controller.py .                                [ 23%]
api/api/controllers/test/test_rootcheck_controller.py ....                            [ 23%]
api/api/controllers/test/test_rule_controller.py ........                             [ 25%]
api/api/controllers/test/test_sca_controller.py ..                                    [ 25%]
api/api/controllers/test/test_security_controller.py ................................ [ 31%]
...................                                                                   [ 35%]
api/api/controllers/test/test_syscheck_controller.py ....                             [ 35%]
api/api/controllers/test/test_syscollector_controller.py .........                    [ 37%]
api/api/controllers/test/test_task_controller.py .                                    [ 37%]
api/api/controllers/test/test_vulnerability_controller.py ....                        [ 38%]
api/api/models/test/test_model.py ...........................                         [ 43%]
api/api/test/test_alogging.py ..................                                      [ 46%]
api/api/test/test_authentication.py ...........                                       [ 48%]
api/api/test/test_configuration.py ..........................................         [ 56%]
api/api/test/test_encoder.py ...                                                      [ 57%]
api/api/test/test_middlewares.py ............                                         [ 59%]
api/api/test/test_uri_parser.py ...                                                   [ 60%]
api/api/test/test_util.py ........................................                    [ 67%]
api/api/test/test_validator.py ...................................................... [ 77%]
..................................................................................... [ 93%]
.....................................                                                 [100%]

============================= 540 passed, 19 warnings in 2.32s ==============================
```

</details>

<details> <summary> Framework  UT </summary>

```
(unit-test) eduardoleon@pop-os:~/git/wazuh$ pytest --disable-warnings framework/
==================================== test session starts ====================================
platform linux -- Python 3.9.9, pytest-7.0.1, pluggy-0.13.1
rootdir: /home/eduardoleon/git/wazuh/framework
plugins: asyncio-0.18.1, aiohttp-1.0.4, trio-0.7.0, cov-3.0.0, anyio-3.6.2, metadata-2.0.2, html-3.0.0
asyncio: mode=auto
collected 2049 items

framework/scripts/tests/test_agent_groups.py ..............                           [  0%]
framework/scripts/tests/test_agent_upgrade.py ...............                         [  1%]
framework/scripts/tests/test_cluster_control.py ......                                [  1%]
framework/scripts/tests/test_wazuh_clusterd.py .......                                [  2%]
framework/scripts/tests/test_wazuh_logtest.py ......................                  [  3%]
framework/wazuh/core/cluster/dapi/tests/test_dapi.py ................................ [  4%]
                                                                                      [  4%]
framework/wazuh/core/cluster/tests/test_client.py ................                    [  5%]
framework/wazuh/core/cluster/tests/test_cluster.py .................................. [  7%]
.                                                                                     [  7%]
framework/wazuh/core/cluster/tests/test_common.py ................................... [  8%]
........................................                                              [ 10%]
framework/wazuh/core/cluster/tests/test_control.py ......                             [ 11%]
framework/wazuh/core/cluster/tests/test_local_client.py .............                 [ 11%]
framework/wazuh/core/cluster/tests/test_local_server.py ........................      [ 12%]
framework/wazuh/core/cluster/tests/test_master.py ................................... [ 14%]
..........                                                                            [ 15%]
framework/wazuh/core/cluster/tests/test_server.py ...................                 [ 16%]
framework/wazuh/core/cluster/tests/test_utils.py ...........                          [ 16%]
framework/wazuh/core/cluster/tests/test_worker.py ................................... [ 18%]
.                                                                                     [ 18%]
framework/wazuh/core/tests/test_active_response.py ..............                     [ 19%]
framework/wazuh/core/tests/test_agent.py ............................................ [ 21%]
..................................................................................... [ 25%]
.................                                                                     [ 26%]
framework/wazuh/core/tests/test_cdb_list.py ......................................    [ 28%]
framework/wazuh/core/tests/test_common.py .......                                     [ 28%]
framework/wazuh/core/tests/test_configuration.py .................................... [ 30%]
..................................                                                    [ 31%]
framework/wazuh/core/tests/test_database.py .............                             [ 32%]
framework/wazuh/core/tests/test_decoder.py ................                           [ 33%]
framework/wazuh/core/tests/test_exception.py ...                                      [ 33%]
framework/wazuh/core/tests/test_input_validator.py ...                                [ 33%]
framework/wazuh/core/tests/test_logtest.py ..                                         [ 33%]
framework/wazuh/core/tests/test_manager.py ...............                            [ 34%]
framework/wazuh/core/tests/test_mitre.py .............                                [ 34%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                               [ 35%]
framework/wazuh/core/tests/test_results.py ........................................   [ 37%]
framework/wazuh/core/tests/test_rootcheck.py .............                            [ 37%]
framework/wazuh/core/tests/test_rule.py ......................                        [ 38%]
framework/wazuh/core/tests/test_sca.py ........................                       [ 40%]
framework/wazuh/core/tests/test_security.py ............                              [ 40%]
framework/wazuh/core/tests/test_stats.py .................                            [ 41%]
framework/wazuh/core/tests/test_syscheck.py .......                                   [ 41%]
framework/wazuh/core/tests/test_syscollector.py ...                                   [ 41%]
framework/wazuh/core/tests/test_task.py ........                                      [ 42%]
framework/wazuh/core/tests/test_utils.py ............................................ [ 44%]
..................................................................................... [ 48%]
..................................................................................... [ 52%]
............................                                                          [ 54%]
framework/wazuh/core/tests/test_vulnerability.py ..                                   [ 54%]
framework/wazuh/core/tests/test_wazuh_queue.py ....................                   [ 55%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................                  [ 56%]
framework/wazuh/core/tests/test_wdb.py ...............................                [ 57%]
framework/wazuh/core/tests/test_wlogging.py .........                                 [ 58%]
framework/wazuh/rbac/tests/test_auth_context.py ..                                    [ 58%]
framework/wazuh/rbac/tests/test_decorators.py ....................................... [ 60%]
......................................................................                [ 63%]
framework/wazuh/rbac/tests/test_default_configuration.py ............................ [ 64%]
............................                                                          [ 66%]
framework/wazuh/rbac/tests/test_orm.py .............................................. [ 68%]
........                                                                              [ 68%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........                           [ 69%]
framework/wazuh/tests/test_active_response.py ............                            [ 70%]
framework/wazuh/tests/test_agent.py ................................................. [ 72%]
....................................................................                  [ 75%]
framework/wazuh/tests/test_cdb_list.py .............................................. [ 77%]
.......                                                                               [ 78%]
framework/wazuh/tests/test_ciscat.py .................................                [ 79%]
framework/wazuh/tests/test_cluster.py ..........                                      [ 80%]
framework/wazuh/tests/test_decoder.py ...................................             [ 82%]
framework/wazuh/tests/test_group.py .......                                           [ 82%]
framework/wazuh/tests/test_logtest.py ......                                          [ 82%]
framework/wazuh/tests/test_manager.py ....................................            [ 84%]
framework/wazuh/tests/test_mitre.py .......                                           [ 84%]
framework/wazuh/tests/test_rootcheck.py ............................................. [ 87%]
.....                                                                                 [ 87%]
framework/wazuh/tests/test_rule.py .................................................. [ 89%]
......                                                                                [ 90%]
framework/wazuh/tests/test_sca.py ...........                                         [ 90%]
framework/wazuh/tests/test_security.py .............................................. [ 92%]
...........................                                                           [ 94%]
framework/wazuh/tests/test_stats.py ...............                                   [ 94%]
framework/wazuh/tests/test_syscheck.py .........................                      [ 96%]
framework/wazuh/tests/test_syscollector.py ............                               [ 96%]
framework/wazuh/tests/test_task.py ............................                       [ 98%]
framework/wazuh/tests/test_vulnerability.py ........................................  [100%]

======================= 2049 passed, 50 warnings in 238.30s (0:03:58) =======================
```

</details>

### CODE REVIEW

File: api/api/validator.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
 _sort_param = re.compile(r'^[\w_\-,\s+.]+$')
 _timeframe_type = re.compile(r'^(\d+[dhms]?)$')
 _type_format = re.compile(r'^xml$|^json$')
+_wpk_path = re.compile(r'^[\w\-.\\/]+(\/[\w\-.\\/]+)*\/[\w\-.]+\.[\w]+$')
```
Comment (fdalmaup) [inline api/api/validator.py:51]: Since a new regex is being added, test cases should also be added to `test_validator.py`

**Review by fdalmaup — CHANGES_REQUESTED** [review]
Review the regex since the `/var/ossec/correct_file.txt` file_path is being taken into account

File: api/api/validator.py (hunk 1 above)
Comment (EduLeon12) [reply api/api/validator.py]: Done !!

Comment (EduLeon12) [reply api/api/validator.py]: Done. 
Description updated with results.

File: api/api/test/test_validator.py [hunk 2]
```diff
@@ -87,7 +87,8 @@
     ('wazuh 4.4.0', _wazuh_version),
     ('wazuh v4.4.0', _wazuh_version),
     # miscellaneous
-    ('aHR0cHM6Ly9zdGFja2FidXNlLmNvbS90YWcvamF2YS8=', _base64)
+    ('aHR0cHM6Ly9zdGFja2FidXNlLmNvbS90YWcvamF2YS8=', _base64),
+    ('correct.wpk', _wpk_path)
```
Comment (fdalmaup) [inline api/api/test/test_validator.py:91]: The test case could be under the `#paths`  comment, like the `('local_rules1.xml,local_rules2.xml', _xml_filename)` case to maintain the organization of the test cases

File: api/api/test/test_validator.py [hunk 3]
```diff
@@ -159,7 +160,9 @@ def test_validation_check_exp_ok(exp, regex_name):
     ('wazuh 4.4', _wazuh_version),
     ('wazuh v4.4', _wazuh_version),
     # miscellaneous
-    ('aDhjasdh3=', _base64)
+    ('aDhjasdh3=', _base64),
+    ('incorrect.txt', _wpk_path),
+    ('.wpk', _wpk_path)
```
Comment (fdalmaup) [inline api/api/test/test_validator.py:165]: Same as the previous comment.

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: api/api/test/test_validator.py (hunk 2 above)
Comment (EduLeon12) [reply api/api/test/test_validator.py]: Done.

File: api/api/test/test_validator.py (hunk 3 above)
Comment (EduLeon12) [reply api/api/test/test_validator.py]: Also Done

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Review by Selutario — APPROVED** [review]

**Pull request merged**